### DISCUSSION TITLE
Shoud I increase CPU count?

### DISCUSSION THREAD

**Discussion by lee-costa** [blakeblackshear/frigate#7839] [General]
I am running Frigate in a VM under ESXi 7 and while it works fantastically well (I am using a Coral passed through the VM), my CPU (E3-1245 v6) doesn't allow for hardware acceleration as recommended so I am stuck using CPU. My VM has 3 vCPUs configured and 4GB of memory. Frigate and a couple other containers (Compreface and Doubletake) also runs on this VM. 

While inspecting my CPU usage, I've noticed the following when running htop under Ubuntu:

![Screen Shot 2023-09-16 at 1 06 48 PM](https://github.com/blakeblackshear/frigate/assets/22748735/041b2e39-a6aa-472d-8275-c87dfdcf0ccb)

It appears that each CPU is sharing the load nicely (correct me if I am wrong) but I wonder if Frigate would benefit if I were to increase CPU count.

I am running 3 cameras all using 4K for detection even though I know it's not recommended but due to the distance they are installed, I needed 4K for a larger field of view due to the substream of my cameras only allowing for 704x480 (Amcrest IP8M-2496EW-V2).

Frigate works well but I was wondering if there's something that can be improved with my setup as far as memory and CPU goes.

Pardon in advance for the question but I know you folks know your stuff so ;-)

**Comment by NickM-27** [comment]
> I am running 3 cameras all using 4K for detection even though I know it's not recommended but due to the distance they are installed, I needed 4K for a larger field of view due to the substream of my cameras only allowing for 704x480 (Amcrest IP8M-2496EW-V2).

you would probably find better results scaling that down to a lower resolution using frigate detect config. This will of course also use cpu but I think you might find low resource usage and better results running on something more reasonable

> Frigate works well but I was wondering if there's something that can be improved with my setup as far as memory and CPU goes.

the load could be spread out over more cpus, I don't think the behavior of frigate itself would improve with more CPUs in your case currently

**Comment by lee-costa** [comment]
I will try lowering the resolution to something more manageable which again I assume to be something that fits my needs/area being monitored?

Thanks for your input.

Edit - I deleted my last commend since I was logged into the wrong account 😑.

**Reply by lee-costa** [reply]
I tried configuring detect to 1920x1080 and configured my required zones to be nearly the same and for some reason detect is a lot more sensitive picking up my parked car when it din’t before (only if it moved which was the goal) and motion boxes are constantly on the video feed when viewing in debug mode. 

I can post my configuration but why is this happening exactly? Simply due to the resolution change?

**Reply by NickM-27** [reply]
configuration would be good to see, generally I would suggest a resolution of 1280x720. Changing the resolution does affect the default motion settings, it is possible that the scaling down using the software (CPU) decoder is introducing noise which is being seen as motion. Not 100% sure though, I've not seen that issue using GPU decoder

**Reply by lee-costa** [reply]
My configuration is below (only for the camera in question). I may end up hanging on to the 4K configuration due to the complexity of my setup since it works so well. It would be nice to gain some CPU cycles back though. 

        #########################################################
        ## Global
        #########################################################

        mqtt:
          host: 192.168.141.21
          user: lee
          password: REDACTED

        birdseye:
          enabled: True
          mode: continuous

        detectors:
          coral:
            type: edgetpu
            device: usb

        database:
          path: /db/frigate.db

        snapshots:
          enabled: True
          timestamp: False
          bounding_box: True
          crop: False
          required_zones: []
          retain:
            default: 21

        go2rtc:
          streams:
            driveway_cam:
              - rtsp://{FRIGATE_RTSP_USER}:{FRIGATE_RTSP_PASSWORD}@192.168.141.14:554/cam/realmonitor?channel=1&subtype=0
            backyard_cam:
              - rtsp://{FRIGATE_RTSP_USER}:{FRIGATE_RTSP_PASSWORD}@192.168.141.15:554/cam/realmonitor?channel=1&subtype=0
            living_room_cam:
              - rtsp://{FRIGATE_RTSP_USER}:{FRIGATE_RTSP_PASSWORD}@192.168.141.16:554/cam/realmonitor?channel=1&subtype=0

        ##########################################################
        ## Driveway
        ##########################################################

        cameras:
          driveway_cam:
            ffmpeg:
              output_args:
                record: preset-record-generic-audio-copy
              inputs:
                - path: rtsp://127.0.0.1:8554/driveway_cam # <--- the name here must match the name of the camera in restream
                  input_args: preset-rtsp-restream
                  roles:
                    - record
                    - detect

            detect:
              width: 1920
              height: 1080
              fps: 5
              enabled: True

            zones:
              driveway:
                coordinates: 0,2160,0,1765,2067,1347,3840,1796,3840,2160
                objects:
                  - person
                  - animal
                  - motorcycle
                  - bicycle
              street:
                coordinates: 2528,1432,2806,1297,3840,1545,3840,1761
                objects:
                  - person
                  - animal
                  - car
                  - motorcycle
                  - bicycle

            objects:
              track:
                - person
                - animal
                - car
                - motorcycle
                - bicycle
              filters:
                person:
                  min_area: 40000
                  max_area: 100000
                  mask:
                    - 0,0,0,2160,0,1661,0,1765,2470,1223,3840,1537,3840,0
                car:
                  min_area: 5000
                  max_area: 200000
                  mask:
                    - 0,0,0,2160,0,1661,0,1765,2470,1223,3840,1537,3840,0
                animal:
                  min_area: 1000
                  mask:
                    - 0,0,0,2160,0,1661,0,1765,2470,1223,3840,1537,3840,0
                motorcycle:
                  mask:
                    - 0,0,0,2160,0,1661,0,1765,2470,1223,3840,1537,3840,0
                bicycle:
                  mask:
                    - 0,0,0,2160,0,1661,0,1765,2470,1223,3840,1537,3840,0

            motion:
              mask:
              - 0,0,0,2160,0,1661,0,1765,2470,1223,3840,1537,3840,0

            record:
              enabled: True
              retain:
                days: 21
                mode: motion
              events:
                retain:
                  default: 21
                  mode: active_objects
                required_zones:
                  - driveway
                  - street

            snapshots:
              enabled: True
              bounding_box: True
              required_zones:
                - driveway
                - street
              retain:
                default: 21

**Reply by NickM-27** [reply]
not sure if this is just due to the changes but I notice the masks and zones are not changed to fit this resolution

**Reply by lee-costa** [reply]
It must of been the wrong file from my clipboard. Disregard the masks.
